src/plugins/global-settings/preset-controls/index.php

```diff
@@ -0,0 +1,172 @@
+			),
+			'spacingSizes' => array(
+				'settings' => array( 'spacing', 'spacingSizes' ),
+				'prefix' => 'spacing',
+			),
+			'blockHeights' => array(
+				'settings' => array( 'blockHeights' ),
+				'prefix' => 'block-height',
+			),
+			'borderRadius' => array(
+				'settings' => array( 'borderRadius' ),
+				'prefix' => 'border-radius',
+			),
+		);
+
+		public $custom_presets;
+		public $theme_presets;
+		public $default_presets;
+		public $stackable_presets;
+
+		/**
+		 * Initialize
+		 */
+  		function __construct() {
+			$this->custom_presets = get_option( 'stackable_global_custom_preset_controls' );
+			$this->theme_presets = WP_Theme_JSON_Resolver::get_theme_data()->get_settings();
+			$this->default_presets = WP_Theme_JSON_Resolver::get_core_data()->get_settings();
+			$this->stackable_presets = $this->load_presets( __DIR__ . '/presets.json');
+
+			add_filter( 'stackable_inline_styles_nodep', array( $this, 'add_preset_controls_styles' ) );
+			add_filter( 'stackable_inline_editor_styles', array( $this, 'add_preset_controls_styles' ) );
+		}
+
+		public static function sanitize_array_setting( $input ) {
+			return ! is_array( $input ) ? array( array() ) : $input;
+		}
+
+		private function load_presets( $json_path ) {
+			if ( file_exists( $json_path ) ) {
+				$json_data = file_get_contents( $json_path );
+				$decoded_data = json_decode( $json_data, true );
+				return $decoded_data[ 'settings' ] ?? [];
+			}
+			return [];
+		}
+
+		/**
+		 * Generate CSS variables based on the property (e.g., fontSizes, spacing).
+		 * The given presets will be overriden it match with a preset from custom.
+		 * 
+		 * @param array $property 
+		 * @param array $presets 
+		 * @param array $prefix 
+		 * @param bool $isTheme
+		 * @return mixed
+		 */
+		public function generate_css_variables( $property, $presets, $prefix, $isTheme = false ) {
+			$filter_name =  current_filter();
+			$custom_presets = $this->custom_presets[ $property ] ?? [];
+			
+			$css = "";
+
+			
+			$presets_by_slug = [];
+			// Convert presets into an associative array with key 'slug'
+			foreach ( $presets as $preset ) {
+				$presets_by_slug[ $preset[ 'slug' ] ] = $preset;
+			}
+
+			if ( $filter_name !== 'stackable_inline_editor_styles' ) {
+				// Override values in base presets if it exist in custom presets
+				foreach ( $custom_presets as $custom ) {
+					$custom[ '__is_custom' ] = true;
+					$presets_by_slug[ $custom[ 'slug' ] ] = $custom;
+				}
+			}
+
+			// If custom presets or using stackable presets, use the given size.
+			// If using theme presets, use WP generated --wp-preset to support theme.json specific
+			// configuration (fluid, clamping, etc.)
+			foreach ( $presets_by_slug as $slug => $preset ) {
+				$is_custom = $preset[ '__is_custom' ] ?? false;
+		
+				$value = $is_custom || ! $isTheme
+					? $preset['size']
+					: "var(--wp--preset--$prefix--$slug)";
+		
+				$css .= "--stk--preset--$prefix--$slug: $value;\n";
```

Comment:
Instead of building the css from strings, you can return an object here instead, then later on use `wp_style_engine_get_stylesheet_from_css_rules` to build the css so we get the benefit of sanitation.